packages/tuono/src/types.ts

```diff
@@ -1,3 +1,52 @@
+/**
+ * Provided from the the rust server and use in ssr env
+ * @see tuono-router {@link ServerInitialLocation}
+ */
+export interface ServerPayloadLocation {
+  href: string
+  pathname: string
+  searchStr: string
+}
+
+/**
+ * @see crates/tuono_lib/src/payload.rs
+ * @warning keep in sync with the same interface inside tuono-router until router is not specialized
+ */
+export interface ServerPayload<TData = unknown> {
+  mode: 'Prod' | 'Dev'
+
+  location: ServerPayloadLocation
+
+  data: TData
+
+  /** Available only on 'Prod' mode */
+  jsBundles: Array<string> | null
+  cssBundles: Array<string> | null
+
+  /** Available only on 'Dev' mode */
+  devServerConfig?: {
+    port: number
+    host: string
+  }
+}
+
+/* the above type could be refined with an union like this
+(
+  | {
+      mode: 'Prod'
+      jsBundles: Array<string>
+      cssBundles: Array<string>
+    }
+  | {
+      mode: 'Dev'
+      devServerConfig: {
+        port: number
+        host: string
+      }
+    }
+)
+*/
```

Comment:
Another type checking refinements that might better describe the runtime behaviour of payload from server. Also this one should be addressed in a different PR.

Reply:
I like the union Idea

Reply:
https://github.com/orgs/tuono-labs/projects/2?pane=issue&itemId=95174765